Support matplotlib>=3.6

Currently some of the plotting tests fail
```shell
====================================================================== FAILURES =======================================================================
____________________________________________________________ test_plot_2D_line_rect_region ____________________________________________________________
Traceback (most recent call last):
  File "/home/vid18871/projects/scanspec/tests/test_cli.py", line 148, in test_plot_2D_line_rect_region
    assert result.exit_code == 0
AssertionError: assert 1 == 0
 +  where 1 = <Result MatplotlibDeprecationWarning('Passing the angle parameter of __init__() positionally is deprecated since Matplotlib 3.6; the parameter will become keyword-only two minor releases later.')>.exit_code
__________________________________________________________________ test_plot_3D_line __________________________________________________________________
Traceback (most recent call last):
  File "/home/vid18871/projects/scanspec/tests/test_cli.py", line 216, in test_plot_3D_line
    assert len(arrow_artists) == 4
AssertionError: assert 0 == 4
 +  where 0 = len([])
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -29,7 +29,7 @@ requires-python = ">=3.8"
 plotting = [
     # make sure a python 3.9 compatible scipy and matplotlib are selected
     "scipy>=1.5.4",
-    "matplotlib>=3.2.2,<3.6",
+    "matplotlib>=3.2.2",
 ]
 # GraphQL service support
 graphql = ["graphql-server[aiohttp]==3.0.0b5", "aiohttp_cors"]

--- a/src/scanspec/plot.py
+++ b/src/scanspec/plot.py
@@ -52,7 +52,6 @@ def _plot_arrow(axes, arrays: List[np.ndarray]):
         a = _Arrow3D(
             *arrows[:3], mutation_scale=10, arrowstyle="-|>", color="lightgrey"
         )
-
         axes.add_artist(a)
 
 

--- a/src/scanspec/plot.py
+++ b/src/scanspec/plot.py
@@ -52,6 +52,7 @@ def _plot_arrow(axes, arrays: List[np.ndarray]):
         a = _Arrow3D(
             *arrows[:3], mutation_scale=10, arrowstyle="-|>", color="lightgrey"
         )
+
         axes.add_artist(a)
 
 
@@ -132,11 +133,11 @@ def plot_spec(spec: Spec[Any]):
                 width = region.x_max - region.x_min
                 height = region.y_max - region.y_min
                 plt_axes.add_patch(
-                    patches.Rectangle(xy, width, height, region.angle, fill=False)
+                    patches.Rectangle(xy, width, height, angle=region.angle, fill=False)
                 )
             elif isinstance(region, Circle):
                 xy = (region.x_middle, region.y_middle)
-                plt_axes.add_patch(patches.Circle(xy, region.radius, fill=False))
+                plt_axes.add_patch(patches.Circle(xy, angle=region.radius, fill=False))
             elif isinstance(region, Ellipse):
                 xy = (region.x_middle, region.y_middle)
                 width = region.x_radius * 2

--- a/src/scanspec/plot.py
+++ b/src/scanspec/plot.py
@@ -136,7 +136,7 @@ def plot_spec(spec: Spec[Any]):
                 )
             elif isinstance(region, Circle):
                 xy = (region.x_middle, region.y_middle)
-                plt_axes.add_patch(patches.Circle(xy, angle=region.radius, fill=False))
+                plt_axes.add_patch(patches.Circle(xy, region.radius, fill=False))
             elif isinstance(region, Ellipse):
                 xy = (region.x_middle, region.y_middle)
                 width = region.x_radius * 2

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -210,8 +210,14 @@ def test_plot_3D_line() -> None:
     assert_min_max_3d(lines[12], 0.5, 0.5, 2, 2, 6, 6)
     # Arrows
     extra_artists = axes.get_children()
+
     arrow_artists = list(
-        filter(lambda artist: isinstance(artist, _Arrow3D), extra_artists)
+        filter(
+            lambda artist: isinstance(artist, _Arrow3D)
+            and artist.get_visible()
+            and artist.get_in_layout(),
+            extra_artists,
+        )
     )
     assert len(arrow_artists) == 4
     assert_3d_arrow(arrow_artists[0], 0.5, 2, 5)

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -209,7 +209,7 @@ def test_plot_3D_line() -> None:
     # End
     assert_min_max_3d(lines[12], 0.5, 0.5, 2, 2, 6, 6)
     # Arrows
-    extra_artists = axes.get_default_bbox_extra_artists()
+    extra_artists = axes.get_children()
     arrow_artists = list(
         filter(lambda artist: isinstance(artist, _Arrow3D), extra_artists)
     )

